source/segment_context_impl.cc

```diff
@@ -67,25 +68,57 @@ SpanObject CurrentSegmentSpanImpl::createSpanObject() {
     *entry = log;
   }
 
-  if (parent_segment_context_->parentSpanContextExtension() != nullptr) {
-    if (parent_segment_context_->parentSpanContextExtension()->tracingMode() ==
+  if (parent_segment_context_.parentSpanContextExtension() != nullptr) {
+    if (parent_segment_context_.parentSpanContextExtension()->tracingMode() ==
         TracingMode::Skip) {
       obj.set_skipanalysis(true);
     }
   }
   return obj;
 }
 
-void CurrentSegmentSpanImpl::addLog(int64_t time, std::string& key,
-                                    std::string& value) {
+void CurrentSegmentSpanImpl::addLog(const std::string& key,
+                                    const std::string& value, bool set_time) {
+  assert(!finished_);
   Log l;
-  l.set_time(time);
+  if (set_time) {
+    SystemTimePoint now = SystemTime::now();
+    l.set_time(millisecondsFromEpoch(now));
+  }
   auto* entry = l.add_data();
   entry->set_key(key);
   entry->set_value(value);
   logs_.emplace_back(l);
 }
 
+void CurrentSegmentSpanImpl::startSpan(bool set_time) {
+  if (set_time) {
+    std::cout << "hogehoge" << std::endl;
+    auto now = SystemTime::now();
+    start_time_ = millisecondsFromEpoch(now);
+  }
+}
+
+void CurrentSegmentSpanImpl::endSpan(bool set_time) {
+  assert(!finished_);
+  if (set_time) {
+    auto now = SystemTime::now();
+    end_time_ = millisecondsFromEpoch(now);
+  }
+  finished_ = true;
+}
+
+void CurrentSegmentSpanImpl::setComponentId(int32_t component_id) {
+  assert(!finished_);
+
+  // Component ID is reserved on Skywalking spec.
+  // For more details here:
+  // https://github.com/apache/skywalking/blob/master/docs/en/guides/Component-library-settings.md
+  if (9000 <= component_id && component_id < 10000) {
```

Comment:
ID could be < 9000, if there is defined IDs first time proposed in other language.